python/taichi/main.py

```diff
@@ -255,97 +261,318 @@ def make_argument_parser():
+                    zip.write(f)
+        ver = ti.__version__
+        md5 = hashlib.md5()
+        with open('release.zip', "rb") as f:
+            for chunk in iter(lambda: f.read(4096), b""):
+                md5.update(chunk)
+        md5 = md5.hexdigest()
+        commit = ti.core.get_commit_hash()[:8]
+        fn = f'taichi-src-v{ver[0]}-{ver[1]}-{ver[2]}-{commit}-{md5}.zip'
+        import shutil
+        shutil.move('release.zip', fn)
+
+    def convert(self, arguments: list = sys.argv[2:]):
+        """???"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.convert.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        import shutil
+        # http://www.commandlinefu.com/commands/view/3584/remove-color-codes-special-characters-with-sed
+        # TODO: Windows support
+        for fn in sys.argv[2:]:
+            print("Converting logging file: {}".format(fn))
+            tmp_fn = '/tmp/{}.{:05d}.backup'.format(fn,
+                                                    random.randint(0, 10000))
+            shutil.move(fn, tmp_fn)
+            command = r'sed -r "s/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g"'
+            os.system('{} {} > {}'.format(command, tmp_fn, fn))
+
+    def gif(self, arguments: list = sys.argv[2:]):
+        """Convert mp4 file to gif"""
+        parser = argparse.ArgumentParser(description=f"{self.gif.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+        
+        # TODO: Convert the logic to use args
+        input_fn = sys.argv[2]
+        assert input_fn[-4:] == '.mp4'
+        output_fn = input_fn[:-4] + '.gif'
+        ti.info('Converting {} to {}'.format(input_fn, output_fn))
+        framerate = 24
+        mp4_to_gif(input_fn, output_fn, framerate)
+
+    def video_speed(self, arguments: list = sys.argv[2:]):
+        """Speed up video"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.video_speed.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
+
+        if len(sys.argv) != 4:
+            print('Usage: ti video_speed fn speed_up_factor')
             return -1
-        print(sys.argv)
-        name = sys.argv[1]
-        task = ti.Task(name)
-        task.run(*sys.argv[2:])
-    elif mode == "debug":
-        ti.core.set_core_trigger_gdb_when_crash(True)
-        if argc <= 2:
-            print("Please specify [file name], e.g. render.py")
+        input_fn = sys.argv[2]
+        assert input_fn[-4:] == '.mp4'
+        output_fn = input_fn[:-4] + '-sped.mp4'
+        speed = float(sys.argv[3])
+        accelerate_video(input_fn, output_fn, speed)
+
+    def video_crop(self, arguments: list = sys.argv[2:]):
+        """Crop video"""
+        # TODO: Convert the logic to use args
+        parser = argparse.ArgumentParser(description=f"{self.video_crop.__doc__}")
+        args = parser.parse_args(sys.argv[2:])
```

Comment:
```suggestion
        args = parser.parse_args(arguments)
```
?

Reply:
Thanks! I intentionally kept all th logic code blocks unchanged to make the refactor easier to review between commits. Step 2 will cover resolving the TODOs and proper usage of the args.😅